controller/nodegroup.go

```diff
@@ -253,13 +253,27 @@ func createNodeGroup(config *eksv1.EKSClusterConfig, group eksv1.NodeGroup, eksS
 		nodeGroupCreateInput.Subnets = aws.StringSlice(config.Status.Subnets)
 	}
 
-	finalTemplate := fmt.Sprintf(templates.NodeInstanceRoleTemplate, getEC2ServiceEndpoint(config.Spec.Region))
-	output, err := createStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), config.Spec.DisplayName, finalTemplate, []string{cloudformation.CapabilityCapabilityIam}, []*cloudformation.Parameter{})
-	if err != nil {
-		return "", err
+	if aws.StringValue(group.NodeRole) == "" {
+		if config.Status.GeneratedNodeRole == "" {
+			finalTemplate := fmt.Sprintf(templates.NodeInstanceRoleTemplate, getEC2ServiceEndpoint(config.Spec.Region))
+			output, err := createStack(svc, fmt.Sprintf("%s-node-instance-role", config.Spec.DisplayName), config.Spec.DisplayName, finalTemplate, []string{cloudformation.CapabilityCapabilityIam}, []*cloudformation.Parameter{})
+			if err != nil {
+				// If there was an error creating the node role stack, return an empty launch template
+				// version and the error.
+				return "", err
+			}
+
+			// Ensure that the change persists for the calling function. No need to save here because
+			// the config object will get saved by the calling function.
+			*config = *config.DeepCopy()
```

Comment:
It was experimental to persist the change of `Status.GeneratedNodeRole = <value>` back to the calling function, but it's not working from what I've tested so I updated the fix to return the generated node role to the caller function. The same thing is actually being done with other fields set on the status (managedlaunchtemplateId, managedLaunchTemplateVersions) so it follows design. I still have to test it with the set of updates from rancher/rancher

Reply:
Ok, so I was testing without an updated eks crd.

Once I fixed that, the deepCopy here does work. It persists the change to `config.Status` back to the calling function instead of the change being lost. I realized it's also easier to set the status in nodegroup.go, because we are checking the value of `GeneratedNodeRole` here. I've updated the fix

Reply:
I don't think the issue is whether or not the config object gets copied, but rather the `DeepCopy` call doesn't actually do anything useful. As @rmweir pointed out, `DeepCopy` is used when we don't want to modify the original data being pointed to, in this case we just do an in place object copy so we aren't actually doing anything useful.

Reply:
Again, I don't think this does anything and believe it should be removed if there is no justification for keeping it.